### PR TITLE
153 browser

### PR DESCRIPTION
**Pull request by chStaiger**
- Add a button to upload a Folder in the browser
- Fix crash during login #154

### CODE REVIEW

File: ibridgesgui/browser.py [hunk 1]
```diff
@@ -122,26 +123,22 @@ def create_collection(self):
         coll_widget.exec()
         self.load_browser_table()
 
+    def folder_upload(self):
+        """Select a folder and upload"""
+        select_dir = PyQt6.QtWidgets.QFileDialog.getExistingDirectory(self, "Select Directory")
+        path = self._fs_select(select_dir)
+        #print(path)
```
Comment (qubixes) [inline ibridgesgui/browser.py:130]:
```suggestion
```

File: ibridgesgui/browser.py [hunk 2]
```diff
@@ -488,3 +485,41 @@ def _metadata_edits(self, operation):
...
+        no_button = PyQt6.QtWidgets.QMessageBox.StandardButton.No
+        if isinstance(path_select, tuple):
+            path = path_select[0]
+            reply = PyQt6.QtWidgets.QMessageBox.question(self,
+                                     'Upload File', path,
+                                     yes_button | no_button, no_button)
+        else:
+            path = path_select
+            reply = PyQt6.QtWidgets.QMessageBox.question(self,
+                                     'Upload Folder', path,
```
Comment (qubixes) [inline ibridgesgui/browser.py:505]: This dialog is a little bit on the sparse side. (Also the upload file) Perhaps at least tell them their file will be uploaded to this destination?

File: ibridgesgui/browser.py [hunk 3]
```diff
@@ -488,3 +485,41 @@ def _metadata_edits(self, operation):
...
+                                     yes_button | no_button, no_button)
+
+        if reply == yes_button and path != '':
+            return Path(path)
+
+    def _upload(self, source):
+        """Uploads data to path in inputPath"""
+        overwrite = self.overwrite.isChecked()
```
Comment (qubixes) [inline ibridgesgui/browser.py:513]: This is okay for now, but I would make it something that is set when it actually tries to overwrite.

**Review by qubixes — CHANGES_REQUESTED** [review]
The basics work, so that is really nice. Getting to full functionality! Apart from the comments in person, I have some smaller questions/suggestions.

File: ibridgesgui/browser.py (hunk 2 above)
Comment (chStaiger) [reply ibridgesgui/browser.py]: I changed the info to:
```
Upload File:
/Users/staig001/Demo

to
/nluu12p/home/research-test-christine

Only new data will be added.
```
If `overwrite` is checked the last sentence will switch to: `All data will be updated.`

**Review by qubixes — APPROVED** [review]

**Pull request merged**